File: data/code_pairs/fortran/967.f90
subroutine flood(zslice, wet, dry, x, y, dmax, flooded_zslice, im, jm, nbwet, nbdry)
  implicit none
  integer, intent(in) :: im, jm, nbwet, nbdry, dmax
  real(8), dimension(im,jm), intent(in) :: zslice, x, y
  real(8), dimension(im,jm), intent(out) :: flooded_zslice
  integer, dimension(nbwet,2), intent(in) :: wet
  integer, dimension(nbdry,2), intent(in) :: dry
  real(8), dimension(nbwet) :: d
  integer :: n, m, dmin_idx, idry, jdry, iwet, jwet, iclose, jclose
  
  flooded_zslice = zslice
  
  do n = 1, nbdry
    idry = dry(n, 1)
    jdry = dry(n, 2)
    do m = 1, nbwet
      iwet = wet(m, 1)
      jwet = wet(m, 2)
      d(m) = sqrt((x(idry, jdry) - x(iwet, jwet))**2 + (y(idry, jdry) - y(iwet, jwet))**2)
    end do
    dmin_idx = minloc(d, 1)
    iclose = wet(dmin_idx, 1)
    jclose = wet(dmin_idx, 2)
    if (dmax == 0 .or. d(dmin_idx) <= dmax) then
      flooded_zslice(idry, jdry) = zslice(iclose, jclose)
    end if
  end do
end subroutine flood

File: data/code_pairs/cpp/967.cpp
#ifndef FLOOD_H
#define FLOOD_H

#include <vector>
#include <cmath>
#include <algorithm>

void flood(const std::vector<std::vector<double>>& zslice,
           const std::vector<std::vector<int>>& wet,
           const std::vector<std::vector<int>>& dry,
           const std::vector<std::vector<double>>& x,
           const std::vector<std::vector<double>>& y,
           double dmax,
           std::vector<std::vector<double>>& flooded_zslice,
           int im, int jm, int nbwet, int nbdry) {
    
    flooded_zslice = zslice;
    std::vector<double> d(nbwet);
    
    for (int n = 0; n < nbdry; ++n) {
        int idry = dry[n][0] - 1;
        int jdry = dry[n][1] - 1;
        
        for (int m = 0; m < nbwet; ++m) {
            int iwet = wet[m][0] - 1;
            int jwet = wet[m][1] - 1;
            
            d[m] = std::sqrt((x[idry][jdry] - x[iwet][jwet]) * (x[idry][jdry] - x[iwet][jwet]) +
                             (y[idry][jdry] - y[iwet][jwet]) * (y[idry][jdry] - y[iwet][jwet]));
        }
        
        auto it = std::min_element(d.begin(), d.end());
        int dmin_idx = std::distance(d.begin(), it);
        
        if (dmax == 0 || *it <= dmax) {
            int iclose = wet[dmin_idx][0] - 1;
            int jclose = wet[dmin_idx][1] - 1;
            flooded_zslice[idry][jdry] = zslice[iclose][jclose];
        }
    }
}

#endif